vyper/ast/utils.py

```diff
@@ -18,97 +18,25 @@
-    yield ('col_offset', start[1])
-    yield ('lineno', start[0])
-    yield ('node_id', node.node_id)  # type: ignore
-    yield ('source_code', source_code)
-
-    end = node.last_token.end if hasattr(node, 'last_token') else (None, None)  # type: ignore
-    yield ('end_lineno', end[0])
-    yield ('end_col_offset', end[1])
-    if hasattr(node, 'last_token'):
-        start_pos = node.first_token.startpos  # type: ignore
-        end_pos = node.last_token.endpos  # type: ignore
-        yield ('src', f"{start_pos}:{end_pos-start_pos}:{source_id}")
-
-    if isinstance(node, python_ast.ClassDef):
-        yield ('class_type', node.class_type)  # type: ignore
-
-    for field_name in vyper_class.get_slots():
-        if field_name not in vy_ast.BASE_NODE_ATTRIBUTES and hasattr(node, field_name):
-            yield (
-                field_name,
-                parse_python_ast(
-                    source_code=source_code,
-                    node=getattr(node, field_name),
-                    source_id=source_id
-                )
-            )
-
-
-def parse_python_ast(source_code: str,
-                     node: python_ast.AST,
-                     source_id: int = 0) -> vy_ast.VyperNode:
-    if isinstance(node, list):
-        return _build_vyper_ast_list(source_code, node, source_id)
-    if not isinstance(node, python_ast.AST):
-        return node
-
-    class_name = node.__class__.__name__
-    if isinstance(node, python_ast.Constant):
-        if node.value is None or isinstance(node.value, bool):
-            class_name = "NameConstant"
-        elif isinstance(node.value, (int, float)):
-            class_name = "Num"
-        elif isinstance(node.value, str):
-            class_name = "Str"
-        elif isinstance(node.value, bytes):
-            class_name = "Bytes"
-
-    if not hasattr(vy_ast, class_name):
-        raise SyntaxException(f'Invalid syntax (unsupported "{class_name}" Python AST node).', node)
-
-    vyper_class = getattr(vy_ast, class_name)
-    for field_name in vyper_class.only_empty_fields:
-        if field_name in node._fields and getattr(node, field_name, None):
-            raise SyntaxException(
-                f'Invalid Vyper Syntax. "{field_name}" is an unsupported attribute field '
-                f'on Python AST "{class_name}" class.',
-                field_name
-            )
-
-    init_kwargs = _build_vyper_ast_init_kwargs(source_code, node, vyper_class, source_id)
-    return vyper_class(**init_kwargs)
-
 
 def parse_to_ast(source_code: str, source_id: int = 0) -> list:
+    """
+    Parses a vyper source string and generates vyper AST nodes.
+
+    Parameters
+    ----------
+    source_code : str
+        The vyper source code to parse.
+    source_id : int, optional
+        Source id to use in the .src member of each node.
+
+    Returns
+    -------
+    list
+        Generated vyper AST nodes.
```

Comment:
Typed or untyped? Optimized or un-optimized?

Reply:
Expanded the comment.